### DISCUSSION TITLE
Is it possible to use Chia for storing data?

### DISCUSSION THREAD

**Discussion by Zerophase** [Chia-Network/chia-blockchain#5500] [Q&A]
I'm just curious if Chia has the capabilities of storing actual data that has use for others.  If that's so wouldn't it bring the cost of data storage down sharply, and all Chia miners would reap severe financial benefits from their huge HDD farms?

I'm not asking if this is a capability yet, but if it's a possibility the protocol supports.

**Comment by lvcivs** [comment]
As far as I understand it, the plots are filled with random numbers, and for every challenge it checks whether your plot has the right combination of numbers in one plot. So that pretty much rules out using the space for anything else.

**Reply by Zerophase** [reply]
That's true. There'd have to be fork to change the protocol. In a few years we can probably farm on SSDs. This could get interesting.

**Reply by ilyakooo0** [reply]
storing arbitrary data is theoretically difficult since everyone would need to know all of the data that everyone stores in order to verify that some farmer actually stores the data.

The way it is implemented right now – you don't need to have all of the data to verify that a farmer stores all of it. You are able to mathematically prove that they had all of the data to construct a proof.
